### DISCUSSION TITLE
Groups of connected devices (tags?)

### DISCUSSION THREAD

**Discussion by vmakeev** [vmakeev/huawei_mesh_router#5] [Ideas]
In #3, it was proposed to add sensors for the number of devices connected to the guest network. 

Unfortunately, Huawei routers allow guest devices to connect only to the primary router, and not to the entire mesh network,
which makes it impossible to create triggers for the number of guest devices outside the coverage area of ​​the primary router (because guest devices will simply not connect to the network).

Perhaps an alternative would be to create **device groups** at the component level.

|  Group name  |       Devices in the group        |
|--------------|-----------------------------------|
|  home_owners | Michael's phone, Michael's laptop |
|  visitors    | Victoria's phone, Eugene's phone  |

- Michael's phone is connected to the "Garage" router
- Michael's laptop is connected to the "Living room" router
- Victoria's phone is connected to the "Living room" router
- Eugene's phone is connected to the "primary" router

In this scenario, the sensors for the number of connected devices will provide the following attributes:

|               Sensor                        | Attributes and values |
|---------------------------------------------|-----------------------|
| `sensor.huawei_mesh_3_clients_garage`         | `guest_clients`: 0 <br/> `hilink_clients`: 0<br/>`wireless_clients`: 1<br />`lan_clients`: 0<br />`wifi_2_4_clients`: 0<br />`wifi_5_clients`: 1<br />`group_home_owners_clients`: 1 _// Michael's phone_<br />`group_visitors_clients`: 0 |
| `sensor.huawei_mesh_3_clients_living_room`    | `guest_clients`: 0 <br/> `hilink_clients`: 0<br/>`wireless_clients`: 2<br />`lan_clients`: 0<br />`wifi_2_4_clients`: 0<br />`wifi_5_clients`: 2<br />`group_home_owners_clients`: 1 _// Michael's laptop_<br />`group_visitors_clients`: 1 _// Victoria's phone_ |
| `sensor.huawei_mesh_3_clients_primary_router` | `guest_clients`: 0 <br/> `hilink_clients`: 2<br/>`wireless_clients`: 3<br />`lan_clients`: 0<br />`wifi_2_4_clients`: 0<br />`wifi_5_clients`: 3<br />`group_home_owners_clients`: 0<br />`group_visitors_clients`: 1 _// Eugene's phone_ |

The list of groups and the mapping of MAC addresses of devices to specific groups should be placed in the component's settings.

The same device can appear in several groups (or should we call them tags?).

**Comment by vmakeev** [comment]
@CosticaPuntaru something like this might be useful in your automations?

**Reply by CosticaPuntaru** [reply]
yes, the guest_clients is exactly what i wanted, but unfortunately, the Huawei guest does not span over the entire mesh, using the guest network is impractical for my home, lets hope pressure will make Huawei fix this issue in a software update

**Comment by vmakeev** [comment]
Implemented at https://github.com/vmakeev/huawei_mesh_router/releases/tag/v0.7.4
